tools/cli/helpers/checkEnvironment.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import child_process from 'child_process';
+import fs from 'fs';
+import * as envfile from 'envfile';
+import chalk from 'chalk';
+
+/**
+ * Get a list of monorepo tooling version requirements to check which versions we need.
+ *
+ * @returns {object} - a list of monorepo version requirements.
+ */
+export function getVersions() {
+	const versions = envfile.parse( fs.readFileSync( `./.github/versions.sh`, 'utf8' ) );
+	return versions;
+}
+
+/**
+ * Returns the current dev environment's composer version.
+ *
+ * @returns {string} - the composer version of the current dev environment.
+ */
+export async function getComposerVersion() {
+	const composerString = await child_process
+		.spawnSync( 'composer', [ '--version' ] )
+		.stdout.toString()
+		.trim();
+	const composerVersion = composerString.match( /\d+.\d+.\d+/ );
+	return composerVersion[ 0 ];
+}
+
+/**
+ * Compares composer versions and exit if it doesn't match.
+ */
+export async function compareComposerVersion() {
+	const currentComposerVersion = await getComposerVersion();
+	const monorepoComposerVersion = getVersions().COMPOSER_VERSION;
+	if ( currentComposerVersion !== monorepoComposerVersion ) {
```

Comment:
Patch-level differences are ok, we just need to check the major.minor.

We've already got the `semver` package as a dependency, we might just use that to do the comparison. `semver.satisfies( currentComposerVersion, '~' + getVersions().COMPOSER_VERSION )`.

Reply:
[5127afd](https://github.com/Automattic/jetpack/pull/23937/commits/5127afd0d4a0500d9f849bfb993ef3eeab547c8d)